### PR TITLE
Feature/lp14 apartment UI

### PR DESCRIPTION
**Pull request by Usiel-Gandalf**
Apartment Screen UI
In this screen the user can show the account what have to pay, pay that account, watch the status of the repairs in their appartment, like open, in progres and closed and the user can open a new repair and one section where the user can make other actions trought buttons.

https://devaptivist.atlassian.net/browse/LP-14?atlOrigin=eyJpIjoiN2M4OTc0ZGU0ZGEzNDg5NTg1NjA4ZGQxNDE0N2MzZWUiLCJwIjoiaiJ9

Apartment UI

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -57,6 +57,7 @@ dependencies {
     kapt libs.hiltCompiler
 
     implementation project(":core:designsystem")
+    implementation project(":feature:apartment")
```
Comment (melissaquezada94) [inline app/build.gradle]: I think is not necessary import the feature apartment.

File: app/src/main/java/com/iteneum/leasepert/MainActivity.kt [hunk 2]
```diff
@@ -3,17 +3,20 @@ package com.iteneum.leasepert
 import android.os.Bundle
 import androidx.activity.ComponentActivity
 import androidx.activity.compose.setContent
+import com.iteneum.apartment.presentation.ApartmentView
```
Comment (melissaquezada94) [inline app/src/main/java/com/iteneum/leasepert/MainActivity.kt]: I think is not important to call the ApartmentView from the activity so you can delete the import and the function

File: feature/apartment/src/main/java/com/iteneum/apartment/presentation/ApartmentView.kt [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+import androidx.compose.foundation.*
+import com.iteneum.designsystem.theme.LeasePertTheme
+import androidx.compose.foundation.layout.*
+import androidx.compose.material3.*
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.tooling.preview.Preview
```
Comment (melissaquezada94) [inline feature/apartment/src/main/java/com/iteneum/apartment/presentation/ApartmentView.kt]: Please delete the import

File: feature/apartment/src/main/java/com/iteneum/apartment/presentation/ApartmentView.kt [hunk 4]
```diff
@@ -0,0 +1,223 @@
...
+ * in this section the user can their account
+ * [RepairsSection] and [RepairStatusRow], work together for show and open a new repairs application with the button,
+ * the user can watch the Open, In Progress and closed repairs status
+ * [InformationSection] is where the user have a their disposable some buttons for leasing documents, apartment photos and current status
+ *
+ * @author Usiel Filiberto Garcia Jimenez
+ */
+@Composable
+@Preview
+fun ApartmentView() {
```
Comment (melissaquezada94) [inline feature/apartment/src/main/java/com/iteneum/apartment/presentation/ApartmentView.kt:27]: I think you don't need the preview, just the composable function for the view.

File: feature/apartment/src/main/java/com/iteneum/apartment/presentation/ApartmentView.kt [hunk 5]
```diff
@@ -0,0 +1,223 @@
...
+import androidx.compose.ui.tooling.preview.Preview
+import com.iteneum.apartment.R
+import com.iteneum.designsystem.components.LPTitleLarge
+import com.iteneum.designsystem.components.LpGenericChip
+import com.iteneum.designsystem.components.LpOutlinedButton
+import com.iteneum.designsystem.components.LpPaymentsRentCard
+import java.sql.Timestamp
+
+/**
+ * [ApartmentView] The apartment view is the screen where the user watches your payments, repairs, information
```
Comment (melissaquezada94) [inline feature/apartment/src/main/java/com/iteneum/apartment/presentation/ApartmentView.kt]: Please check how should be the documentation. The documentation always should be above the function.

File: feature/apartment/src/main/java/com/iteneum/apartment/presentation/ApartmentView.kt [hunk 6]
```diff
@@ -0,0 +1,223 @@
...
+                    bottom = LeasePertTheme.sizes.minorRegularSize
+                ),
+            horizontalArrangement = Arrangement.SpaceBetween
+        ) {
+            Text(
+                text = stringResource(id = R.string.repairs),
+                color = MaterialTheme.colorScheme.tertiary,
+                fontSize = MaterialTheme.typography.titleSmall.lineHeight
+            )
+            LpGenericChip(label = "+${stringResource(id = R.string.newA)}") {
```
Comment (melissaquezada94) [inline feature/apartment/src/main/java/com/iteneum/apartment/presentation/ApartmentView.kt]: Please delete the `+` and send an icon like parameter to `LpGenericChip` function

File: feature/apartment/src/main/java/com/iteneum/apartment/presentation/ApartmentView.kt [hunk 7]
```diff
@@ -0,0 +1,223 @@
...
+            modifier = Modifier
+                .fillMaxWidth()
+                .padding(
+                    top = LeasePertTheme.sizes.regularSize,
+                    bottom = LeasePertTheme.sizes.minorRegularSize
+                ),
+            horizontalArrangement = Arrangement.SpaceBetween
+        ) {
+            Text(
+                text = stringResource(id = R.string.repairs),
```
Comment (melissaquezada94) [inline feature/apartment/src/main/java/com/iteneum/apartment/presentation/ApartmentView.kt]: Please check the `fontWeight` for each section title

File: feature/apartment/src/main/java/com/iteneum/apartment/presentation/ApartmentView.kt [hunk 8]
```diff
@@ -0,0 +1,223 @@
...
+fun PaymentsSection(modifier: Modifier, month: String, quantityToPay: Double, date: Timestamp) {
+    Column() {
+        Row(
+            modifier = modifier
+                .fillMaxWidth()
+                .padding(bottom = LeasePertTheme.sizes.regularSize),
+            horizontalArrangement = Arrangement.SpaceBetween
+        ) {
+            Text(
+                text = stringResource(id = R.string.payments),
```
Comment (melissaquezada94) [inline feature/apartment/src/main/java/com/iteneum/apartment/presentation/ApartmentView.kt]: Please check the fontWeight for each section title

File: feature/apartment/src/main/java/com/iteneum/apartment/presentation/ApartmentView.kt [hunk 9]
```diff
@@ -0,0 +1,223 @@
...
+            )
+        }
+    }
+}
+
+@Composable
+fun InformationSection() {
+    Column(modifier = Modifier.padding(top = LeasePertTheme.sizes.midSmallSize)) {
+        Text(
+            text = stringResource(id = R.string.information),
```
Comment (melissaquezada94) [inline feature/apartment/src/main/java/com/iteneum/apartment/presentation/ApartmentView.kt]: Please check the fontWeight for each section title

File: feature/apartment/src/main/java/com/iteneum/apartment/presentation/ApartmentView.kt [hunk 10]
```diff
@@ -0,0 +1,223 @@
...
+                    top = LeasePertTheme.sizes.extraSize10
+                )
+                .size(LeasePertTheme.sizes.largeSize)
+                .clip(MaterialTheme.shapes.extraLarge)
+                .background(color = MaterialTheme.colorScheme.secondaryContainer),
+            contentAlignment = Alignment.Center
+        ) {
+            Text(
+                text = quantityStatusRepairs.toString(),
+                color = MaterialTheme.colorScheme.onPrimary
```
Comment (melissaquezada94) [inline feature/apartment/src/main/java/com/iteneum/apartment/presentation/ApartmentView.kt]: Please check the color and the font weight for the text number

File: feature/apartment/src/main/res/values/strings.xml [hunk 11]
```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
+    <string name="open">Open</string>
```
Comment (melissaquezada94) [inline feature/apartment/src/main/res/values/strings.xml]: please be more descriptive in all the strings name like: 
`text_status_open`
`text_title_information`
`text_button_history`
`text_button_plus_new`

File: app/src/main/java/com/iteneum/leasepert/MainActivity.kt [hunk 12]
```diff
@@ -3,17 +3,20 @@ package com.iteneum.leasepert
...
 import com.iteneum.designsystem.theme.LeasePertTheme
 import dagger.hilt.android.AndroidEntryPoint
 
+
```
Comment (carlosph816) [inline app/src/main/java/com/iteneum/leasepert/MainActivity.kt]: Delete this space

File: feature/apartment/src/main/java/com/iteneum/apartment/presentation/ApartmentView.kt [hunk 13]
```diff
@@ -0,0 +1,223 @@
...
+            label = stringResource(
+                id = R.string.apartment
+            ),
+            color = MaterialTheme.colorScheme.onPrimaryContainer
+        )
+    }
+}
+
+@Composable
+fun PaymentsSection(modifier: Modifier, month: String, quantityToPay: Double, date: Timestamp) {
```
Comment (carlosph816) [inline feature/apartment/src/main/java/com/iteneum/apartment/presentation/ApartmentView.kt]: We can put always the params as stair style

File: app/build.gradle (hunk 1 above)
Comment (Usiel-Gandalf) [reply app/build.gradle]: Done.

File: app/src/main/java/com/iteneum/leasepert/MainActivity.kt (hunk 2 above)
Comment (Usiel-Gandalf) [reply app/src/main/java/com/iteneum/leasepert/MainActivity.kt]: Done.

File: feature/apartment/src/main/java/com/iteneum/apartment/presentation/ApartmentView.kt (hunk 3 above)
Comment (Usiel-Gandalf) [reply feature/apartment/src/main/java/com/iteneum/apartment/presentation/ApartmentView.kt]: Done.

File: feature/apartment/src/main/java/com/iteneum/apartment/presentation/ApartmentView.kt (hunk 4 above)
Comment (Usiel-Gandalf) [reply feature/apartment/src/main/java/com/iteneum/apartment/presentation/ApartmentView.kt:27]: Done.

File: feature/apartment/src/main/java/com/iteneum/apartment/presentation/ApartmentView.kt (hunk 5 above)
Comment (Usiel-Gandalf) [reply feature/apartment/src/main/java/com/iteneum/apartment/presentation/ApartmentView.kt]: Ok i write a documentation for each function

File: feature/apartment/src/main/java/com/iteneum/apartment/presentation/ApartmentView.kt [hunk 14]
```diff
@@ -0,0 +1,223 @@
...
+                    bottom = LeasePertTheme.sizes.smallerSize,
+                    top = LeasePertTheme.sizes.extraSize10
+                )
+                .size(LeasePertTheme.sizes.largeSize)
+                .clip(MaterialTheme.shapes.extraLarge)
+                .background(color = MaterialTheme.colorScheme.secondaryContainer),
+            contentAlignment = Alignment.Center
+        ) {
+            Text(
+                text = quantityStatusRepairs.toString(),
```
Comment (carlosph816) [inline feature/apartment/src/main/java/com/iteneum/apartment/presentation/ApartmentView.kt:219]: I think if you always are going to do a string cast then the params must be string and not Int and we avoid the casting

File: feature/apartment/src/main/java/com/iteneum/apartment/presentation/ApartmentView.kt (hunk 7 above)
Comment (Usiel-Gandalf) [reply feature/apartment/src/main/java/com/iteneum/apartment/presentation/ApartmentView.kt]: Done

File: feature/apartment/src/main/java/com/iteneum/apartment/presentation/ApartmentView.kt (hunk 8 above)
Comment (Usiel-Gandalf) [reply feature/apartment/src/main/java/com/iteneum/apartment/presentation/ApartmentView.kt]: Done

File: feature/apartment/src/main/java/com/iteneum/apartment/presentation/ApartmentView.kt (hunk 9 above)
Comment (Usiel-Gandalf) [reply feature/apartment/src/main/java/com/iteneum/apartment/presentation/ApartmentView.kt]: They are applied automatic with the theme of te design system

File: feature/apartment/src/main/java/com/iteneum/apartment/presentation/ApartmentView.kt (hunk 6 above)
Comment (Usiel-Gandalf) [reply feature/apartment/src/main/java/com/iteneum/apartment/presentation/ApartmentView.kt]: Done

File: feature/apartment/src/main/java/com/iteneum/apartment/presentation/ApartmentView.kt (hunk 14 above)
Comment (Usiel-Gandalf) [reply feature/apartment/src/main/java/com/iteneum/apartment/presentation/ApartmentView.kt:219]: All params now are string

File: feature/apartment/src/main/java/com/iteneum/apartment/presentation/ApartmentView.kt (hunk 10 above)
Comment (Usiel-Gandalf) [reply feature/apartment/src/main/java/com/iteneum/apartment/presentation/ApartmentView.kt]: Done

File: feature/apartment/src/main/res/values/strings.xml (hunk 11 above)
Comment (Usiel-Gandalf) [reply feature/apartment/src/main/res/values/strings.xml]: Done

File: feature/apartment/src/main/java/com/iteneum/apartment/presentation/ApartmentView.kt [hunk 15]
```diff
@@ -0,0 +1,261 @@
...
+ * [month] is the current month what the user can watch in this section
+ * [quantityToPay] is the quantity of the account what the user must to pay
+ * [limitDateToPay] is the limit date what the user have to pay their account
+ * */
+@Composable
+fun PaymentsSection(
+    modifier: Modifier,
+    month: String,
+    quantityToPay: String,
+    limitDateToPay: String,
```
Comment (carlosph816) [inline feature/apartment/src/main/java/com/iteneum/apartment/presentation/ApartmentView.kt]: Delete the ","

File: feature/apartment/src/main/java/com/iteneum/apartment/presentation/ApartmentView.kt [hunk 16]
```diff
@@ -0,0 +1,261 @@
...
+ * [modifier] Modifier is for specify style and params of the card, like for example the width of the component.
+ * [titleStatusRepairs] is the title what show the row, for example "Closed"
+ * [quantityStatusRepairs] is the quantity of the kind of status what the user have it, for example the user can have opened 3 Closed,
+ * 4 In progress etc.
+ * */
+@Composable
+fun RepairStatusRow(
+    modifier: Modifier = Modifier,
+    titleStatusRepairs: String,
+    quantityStatusRepairs: Int = 0,
```
Comment (carlosph816) [inline feature/apartment/src/main/java/com/iteneum/apartment/presentation/ApartmentView.kt]: Delete this too ","

File: feature/apartment/src/main/java/com/iteneum/apartment/presentation/ApartmentView.kt (hunk 15 above)
Comment (Usiel-Gandalf) [reply feature/apartment/src/main/java/com/iteneum/apartment/presentation/ApartmentView.kt]: Done

File: feature/apartment/src/main/java/com/iteneum/apartment/presentation/ApartmentView.kt (hunk 16 above)
Comment (Usiel-Gandalf) [reply feature/apartment/src/main/java/com/iteneum/apartment/presentation/ApartmentView.kt]: Done

**Review by melissaquezada94 — APPROVED** [review]
Well done!

**Review by carlosph816 — APPROVED** [review]
Well done!

**Pull request merged**